src/ucp/core/ucp_ep.c

```diff
@@ -1287,6 +1306,31 @@ static void ucp_ep_discard_lanes_callback(void *request, ucs_status_t status,
     ucs_free(arg);
 }
 
+static ucs_status_t ucp_ep_failed_op(uct_ep_h ep)
+{
+    ucp_ep_failed_arg_t *arg = ucs_derived_of(ep, ucp_ep_failed_arg_t);
+
+    return arg->status;
+}
+
+static ssize_t ucp_ep_failed_bc_op(uct_ep_h ep)
```

Comment:
can we use `ucp_ep_failed_op` for Bcopy too? anyway, we do explicit casting.

Reply:
no, since return type is different size, ucp_ep_failed_op would set only lower portion of the return value register rax on x86_64